### PR TITLE
[IOTDB-4528] Operate Pipe Procedure 

### PR DESCRIPTION
**Pull request by Cpaulyz**
## Description

This PR implement PIPE operation logic managed by procedure :D
* Support CREATE, START, STOP and DROP operation. There is a demo:

* Not support rollback currently although it have been designed. It will be implemented later.
* For more detail, see https://apache-iotdb.feishu.cn/docx/doxcnpfFAOfJJGj6qmDM4VPEDNf chapter 4. There are state and sequence diagrams to help understand the code.
* IoTDBPipeIT support cluster mode currently.

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/consensus/request/ConfigPhysicalPlanType.java [hunk 1]
```diff
@@ -82,6 +82,9 @@ public enum ConfigPhysicalPlanType {
   CreatePipeSink,
   DropPipeSink,
   GetPipeSink,
+  PreCreatePipe,
+  SetPipeStatus,
+  ShowPipe,
```
Comment (HTHou) [inline confignode/src/main/java/org/apache/iotdb/confignode/consensus/request/ConfigPhysicalPlanType.java:87]: Append them to the end for better compatibility?

**Review by HTHou — COMMENTED** [review]
Look good about datanode part.
Just one suggestion, ptal.

**Comment by Cpaulyz** [comment]
@CRZbulabula @MiniSho Please help review the procedure part to see if it conforms to the specification~

**Comment by MiniSho** [comment]
LGTM~

**Review by CRZbulabula — APPROVED** [review]
LGTM

**Pull request merged**